tests/integration/hdf5/test_io.py

```diff
@@ -418,9 +418,80 @@ def setUp(self):
+        # check empty version before write
+        with NWBHDF5IO(self.path, 'w') as io:
+            self.assertIsNone(io.nwb_version)
+        # write the example file
+        with NWBHDF5IO(self.path, 'w') as io:
+            io.write(self.nwbfile)
+        # check behavior for various different version strings
+        for ver in [("2.0.5", (2, 0, 5)),
+                    ("2.0.5-alpha", (2, 0, 5, "alpha")),
+                    ("bad_version", ("bad_version", ))]:
+            # Set version string
+            with File(self.path, mode='a') as io:
+                io.attrs['nwb_version'] = ver[0]
+            # Assert expected result for nwb_version tuple
+            with NWBHDF5IO(self.path, 'r') as io:
+                self.assertTupleEqual(io.nwb_version, ver[1])
+        # check empty version attribute
+        with File(self.path, mode='a') as io:
+            del io.attrs['nwb_version']
+        with NWBHDF5IO(self.path, 'r') as io:
+            self.assertIsNone(io.nwb_version)
+
+    def test_check_nwb_version_ok(self):
+        """Test that opening a current NWBFile passes the version check"""
+        with NWBHDF5IO(self.path, 'w') as io:
+            io.write(self.nwbfile)
+        with NWBHDF5IO(self.path, 'r') as io:
+            self.assertIsNotNone(io.nwb_version)
+            self.assertGreater(io.nwb_version[0], 1)
+            read_file = io.read()
+            self.assertContainerEqual(read_file, self.nwbfile)
+
+    def test_check_nwb_version_missing_version(self):
+        """Test reading of files with missing nwb_version"""
+        # write the example file
+        with NWBHDF5IO(self.path, 'w') as io:
+            io.write(self.nwbfile)
+        # remove the version attribute
+        with File(self.path, mode='a') as io:
+            del io.attrs['nwb_version']
+        # test that reading the file without a version strings fails
+        with self.assertRaisesWith(
+                TypeError,
+                "Missing NWB version in file. The file is not a valid NWB file."):
+            with NWBHDF5IO(self.path, 'r') as io:
+                _ = io.read()
+        # test that reading the file when skipping the version check works
+        with NWBHDF5IO(self.path, 'r') as io:
+            read_file = io.read(skip_version_check=True)
+            self.assertContainerEqual(read_file, self.nwbfile)
+
+    def test_check_nwb_version_old_version(self):
+        """Test reading of files with version less than 2 """
+        # write the example file
+        with NWBHDF5IO(self.path, 'w') as io:
+            io.write(self.nwbfile)
+        # remove the version attribute
+        with File(self.path, mode='a') as io:
+            io.attrs['nwb_version'] = "1.0.5"
+        # test that reading the file without a version strings fails
+        with self.assertRaisesWith(
+                TypeError,
+                "NWB version (1, 0, 5) not supported. PyNWB supports NWB files version 2 and above."):
```

Comment:
Can you make this print "NWB version 1.0.5 not supported. ..." instead? I think the tuple form is not intuitive to most users.

Reply:
Also note that in this form "1.0.4_beta" will print as `(1, 0, '4_beta')`

Reply:
Thanks for the suggestion @rly. I have now updated ``NWBHDF5IO.nwb_version`` to: 1) return both the original version string and the parsed version tuple and 2) added support for ".", "-", and "_" as delimiters so that ``"1.0.4_beta`` is now parsed correctly to ``(1, 0, 4, 'beta')``.  I also updated the error message in ``NWBHDF5IO.read`` to use the original version string as it appears in the file (rather than reconstructing the version string from the tuple) to make sure that all the delimiters of the version string also match in the error message. I also updated the tests accordingly.

Reply:
Looks good. Thanks